src/main/java/org/jsoup/Jsoup.java

```diff
@@ -73,6 +73,22 @@ public static Connection connect(String url) {
         return HttpConnection.connect(url);
     }
 
+    /**
+     * Creates a new {@link Connection} to a URL. Use to fetch and parse a HTML page.
+     * <p>
+     * Use examples:
+     * <ul>
+     *  <li><code>Document doc = Jsoup.connect("http://example.com").userAgent("Mozilla").data("name", "jsoup").get();</code></li>
+     *  <li><code>Document doc = Jsoup.connect("http://example.com").cookie("auth", "token").post();</code></li>
+     * </ul>
+     * @param url URL to connect to. The protocol must be {@code http} or {@code https}.
+     * @param toPunycode a flag if you need to encode the URL into Punycode in order to connect to IDN (Internationalized Domain Names).
+     * @return the connection. You can add data, cookies, and headers; set the user-agent, referrer, method; and then execute.
+     */
+    public static Connection connect(String url, Boolean toPunycode) {
+        return HttpConnection.connect(url, toPunycode);
```

Comment:
We shouldn't have this method - just punycode all URLs as they come in. Otherwise the user has to know that it's an IDN in the URL for every call, and remember this, and all the link following and redirect code will fail.

Reply:
removed